library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentSync.java

```diff
@@ -57,8 +56,26 @@ private RegistryEntryAttachmentSync() {
 
 	public static final Identifier PACKET_ID = id("sync");
 
-	public record CacheEntry(Identifier registryId,
-	                         Set<Pair<String, NbtCompound>> valueMaps) {
+	private record CacheEntry(Identifier registryId, Set<Pair<String, Set<AttachmentEntry>>> namespacesToValues) {
+	}
+
+	private record AttachmentEntry(String path, boolean isTag, NbtElement value) {
```

Comment:
Is there any code here to guard against an outdated QSL on one side?
A `version` field on the packet might be wise.

Reply:
This is a larger issue than i think this PR should solve. I can add a byte, but will it be properly updated in the future?